mlibrary/design-system
# Responsive Table

# Overview
To add to the [HTML & CSS Workshop](http://localhost:8080/), we now have a responsive table! The design components are  also available on [Figma](https://www.figma.com/file/dxzvdBeoQTcBpSyMoLVQfY/?node-id=444%3A84).

## Setup
To apply the table styles, add the `.m-table` class directly to a `table` element.

### Responsive Breakpoints
There are two methods to make each `table.m-table` responsive:
1. Manually
2. Automatically

#### Manually
For a more universal layout change, there are three classes that can be added to each `table` trigger the responsive layout at specific breakpoints:
| Class                         | Size |
|-------------------------------|-----------|
| `.m-table--responsive-small`  | <= 720px     |
| `.m-table--responsive-medium` | <= 920px     |
| `.m-table--responsive-large`  | <= 1200px    |

These breakpoints were taken from the [Library's reusable media queries](https://github.com/mlibrary/lib.umich.edu/blob/master/src/reusable/media-queries.js).

#### Automatically
Optional JavaScript has been written to automatically make each `table.m-table` responsive at its smallest width. When the width of its parent is smaller than the width of the table, the `.m-table--responsive` class will be added to trigger the layout change. If the parent is not smaller, the class will be removed.

### Responsive Headers
In order for a header to appear in the responsive layout, the cell must have a `data-th` attribute. The value of the attribute should match the text in its associated `thead th` cell.

## Design Tokens
The `extrabold` font weight has been added as a design token, since it is used as the responsive heading, along with the footer.

## Testing
* Run `http://localhost:8080/`
* Check for accessibility
* Check to see if its not broken in...
* Play around with it
  * What does it look like without row headers?
  * Remove the `.m-table--responsive-small` class and see the table become responsive on its own.
  * Add a different responsive class to see where the table breaks.
  * Add or change `data-th` attributes to see how it looks.
